contracts/msig-launcher/src/contract.rs

```diff
@@ -34,216 +32,119 @@ pub fn instantiate(
-                cw4_voting::contract::query,
-            )
-            .with_reply(cw4_voting::contract::reply),
-        ));
-        // Proposal
-        app.store_code(Box::new(
-            ContractWrapper::new(
-                dao_proposal_single::contract::execute,
-                dao_proposal_single::contract::instantiate,
-                dao_proposal_single::contract::query,
-            )
-            .with_reply(dao_proposal_single::contract::reply),
-        ));
-        // Pre Propose
-        app.store_code(Box::new(ContractWrapper::new(
-            dao_pre_propose_single::contract::execute,
-            dao_pre_propose_single::contract::instantiate,
-            dao_pre_propose_single::contract::query,
-        )));
-        // Cw4
-        app.store_code(Box::new(ContractWrapper::new(
-            cw4_group::contract::execute,
-            cw4_group::contract::instantiate,
-            cw4_group::contract::query,
-        )));
+        let mut app = ArchwayApp::default();
+
+        let accounts = app.init_accounts(&[arch(100)], 2).unwrap();
+        let admin = accounts.get(0).unwrap();
+        let user = accounts.get(1).unwrap();
+
+        let wasm = Wasm::new(&app);
+        let dao_core_file = std::fs::read("../../external_wasm/dao_dao_core.wasm").unwrap();
+        wasm.store_code(&dao_core_file, None, admin).unwrap();
+        let voting_file = std::fs::read("../../external_wasm/cw4_voting.wasm").unwrap();
+        wasm.store_code(&voting_file, None, admin).unwrap();
+        let proposal_file = std::fs::read("../../external_wasm/dao_proposal_single.wasm").unwrap();
+        wasm.store_code(&proposal_file, None, admin).unwrap();
+        let pre_propose_file =
+            std::fs::read("../../external_wasm/dao_pre_propose_single.wasm").unwrap();
+        wasm.store_code(&pre_propose_file, None, admin).unwrap();
+        let cw4 = std::fs::read("../../external_wasm/cw4_group.wasm").unwrap();
+        wasm.store_code(&cw4, None, admin).unwrap();
 
         // Msig launcher
-        let launcher_code_id = app.store_code(Box::new(
-            ContractWrapper::new(
-                crate::contract::execute,
-                crate::contract::instantiate,
-                crate::contract::query,
-            )
-            .with_reply(crate::contract::reply),
-        ));
-
-        let launcher = app
-            .instantiate_contract(
+        let launcher =
+            std::fs::read("../../target/wasm32-unknown-unknown/release/msig_launcher.wasm")
+                .unwrap();
+        let launcher_code_id = wasm
+            .store_code(&launcher, None, admin)
+            .unwrap()
+            .data
+            .code_id;
+
+        let launcher = wasm
+            .instantiate(
                 launcher_code_id,
-                Addr::unchecked("admin"),
                 &InstantiateMsg {
                     code_ids: MSigCodeIds {
                         main: 1,
```

Comment:
Minor thing, but would be nicer to use `wasm.store` calls' responses instead of relying on the order.

Reply:
Yeah those are artifacts from the old test, fixed